Assets/Scripts/Behaviours/UnitBehaviour.cs

```diff
@@ -113,10 +113,16 @@ private void OnMouseDown()
                 else if (App.player.ComparePlayerState(PlayerState.placing))
                 {
                     App.unitCardManager.SwitchToCard(null);
-                    App.player.DeleteTransparentUnit(true);
+                    App.player.DeleteTransparentUnit();
+                    App.player.ChangeState(PlayerState.idle);
                 }
             }
         }
+
+        if (App.player.ComparePlayerState(PlayerState.vaccinating))

```

Comment:
Tu by sa tiež malo checkovať !EventSystem.current.IsPointerOverGameObject()
(teoreticky to dať aj s tým hore do nového ifu?)